src/shared/logs-show.c

```diff
@@ -1595,21 +1611,36 @@ int add_matches_for_unit(sd_journal *j, const char *unit) {
         return r;
 }
 
-int add_matches_for_user_unit(sd_journal *j, const char *unit, uid_t uid) {
+int add_matches_for_user_unit_full(sd_journal *j, sd_bus *bus, const char *unit, uid_t uid, bool add_invocation) {
+        sd_id128_t id;
         int r;
 
         assert(j);
         assert(unit);
 
+        if (add_invocation) {
+                assert(bus);
+
+                r = unit_get_invocation_id(bus, unit, &id);
+                if (r < 0)
+                        return r;
+                if (r == 0) {
+                        log_debug("No invocation ID found for %s.", unit);
+                        add_invocation = false;
+                }
+        }
+
         (void) (
                 /* Look for messages from the user service itself */
                 (r = journal_add_match_pair(j, "_SYSTEMD_USER_UNIT", unit)) ||
                 (r = journal_add_matchf(j, "_UID="UID_FMT, uid)) ||
+                (r = add_invocation ? journal_add_match_pair(j, "_SYSTEMD_INVOCATION_ID", SD_ID128_TO_STRING(id)) : 0) ||
 
                 /* Look for messages from systemd about this service */
                 (r = sd_journal_add_disjunction(j)) ||
                 (r = journal_add_match_pair(j, "USER_UNIT", unit)) ||
                 (r = journal_add_matchf(j, "_UID="UID_FMT, uid)) ||
+                (r = add_invocation ? journal_add_match_pair(j, "USER_INVOCATION_ID", SD_ID128_TO_STRING(id)) : 0) ||
 
                 /* Look for coredumps of the service */
                 (r = sd_journal_add_disjunction(j)) ||
```

Comment:
hmm, so afaik coredumps and stuff from "authorized daemons" currently does not include invocation ID stuff in their messages. I think it would make sense to simply drop the coredump and "authorized daemons" block hence for the invocation id stuff

(one reason more to make this separate functions)

Reply:
Well, we actually do have `OBJECT_SYSTEMD_INVOCATION_ID`, but man page entry for that was missing from the beginning... https://github.com/systemd/systemd/pull/32464